Geometry/HcalTowerAlgo/src/HcalGeometry.cc

```diff
@@ -147,19 +147,36 @@ GlobalPoint HcalGeometry::getPosition(const DetId& id) const {
   if (!m_mergePosition) {
     return (getGeometry(id)->getPosition());
   } else {
-    std::vector<HcalDetId> ids;
-    m_topology.unmergeDepthDetId(HcalDetId(id),ids);
-    return (getGeometry(ids.front())->getPosition());
+    std::vector<HcalDetId> ids_;
+    m_topology.unmergeDepthDetId(HcalDetId(id),ids_);
+    return (getGeometry(ids_.front())->getPosition());
   }
 }
 
 GlobalPoint HcalGeometry::getBackPosition(const DetId& id) const {
   if (!m_mergePosition) {
     return (getGeometry(id)->getBackPoint());
   } else {
-    std::vector<HcalDetId> ids;
-    m_topology.unmergeDepthDetId(HcalDetId(id),ids);
-    return (getGeometry(ids.back())->getBackPoint());
+    std::vector<HcalDetId> ids_;
+    m_topology.unmergeDepthDetId(HcalDetId(id),ids_);
+    return (getGeometry(ids_.back())->getBackPoint());
+  }
+}
+
+CaloCellGeometry::CornersVec HcalGeometry::getCorners(const DetId& id) const {
```

Comment:
CaloSubdetectorGeometry does not have getCorners - they are reserved for CaloCellGeometry. This is an additional functionality utilizing CaloCellGeometry functions. I doubt very much if it will break any of the earlier functionality.